pkg/placement/membership.go

```diff
@@ -282,30 +139,34 @@ func (p *Service) processRaftStateCommand(ctx context.Context) {
 				// MemberRemove will be queued by faultHostDetectTimer.
 				// Even if ApplyCommand is failed, both commands will retry
 				// until the state is consistent.
+
 				logApplyConcurrency <- struct{}{}
 				p.wg.Add(1)
 				go func() {
 					defer p.wg.Done()
 
 					// We lock dissemination to ensure the updates can complete before the table is disseminated.
-					p.disseminateLock.Lock()
-					defer p.disseminateLock.Unlock()
+					p.disseminateLocks.Lock(op.host.Namespace)
+					defer p.disseminateLocks.Unlock(op.host.Namespace)
 
-					updated, raftErr := p.raftNode.ApplyCommand(op.cmdType, op.host)
+					updateCount, raftErr := p.raftNode.ApplyCommand(op.cmdType, op.host)
 					if raftErr != nil {
 						log.Errorf("fail to apply command: %v", raftErr)
 					} else {
 						if op.cmdType == raft.MemberRemove {
-							p.lastHeartBeat.Delete(op.host.Name)
+							updateCount = p.handleDisconnectedMember(op, updateCount)
 						}
 
-						// ApplyCommand returns true only if the command changes hashing table.
-						if updated {
-							p.memberUpdateCount.Add(1)
+						// ApplyCommand returns true only if the command changes the hashing table.
+						if updateCount {
+							c, _ := p.memberUpdateCount.GetOrSet(op.host.Namespace, &atomic.Uint32{})
```

Comment:
Are we ignoring errors here?

Reply:
No, this just returns a bool indicating if the value already existed.